ts/nni_manager/training_service/reusable/environments/remoteEnvironmentService.ts

```diff
@@ -248,17 +257,20 @@ export class RemoteEnvironmentService extends EnvironmentService {
             }
             this.environmentExecutorManagerMap.set(environment.id, executorManager);
             const executor = await this.getExecutor(environment.id);
+            let remoteWorkingRoot: string;
             if (environment.useSharedStorage) {
-                const environmentRoot = component.get<SharedStorageService>(SharedStorageService).remoteWorkingRoot;
-                environment.runnerWorkingFolder = executor.joinPath(environmentRoot, 'envs', environment.id)
-                const remoteMountCommand = component.get<SharedStorageService>(SharedStorageService).remoteMountCommand;
-                await executor.executeScript(remoteMountCommand, false, false);
+                remoteWorkingRoot = component.get<SharedStorageService>(SharedStorageService).remoteWorkingRoot;
+                environment.runnerWorkingFolder = executor.joinPath(remoteWorkingRoot, 'envs', environment.id);
+                const remoteMountCommand = component.get<SharedStorageService>(SharedStorageService).remoteMountCommand.replace(/echo -e /g, `echo `).replace(/echo /g, `echo -e `);
+                const result = await executor.executeScript(remoteMountCommand, false, false);
+                if (result.exitCode !== 0) {
+                    throw new Error(`Mount shared storage on remote machine failed.\n ERROR: ${result.stderr}`);
+                }
             } else {
-                environment.runnerWorkingFolder = 
-                    executor.joinPath(executor.getRemoteExperimentRootDir(getExperimentId()), 
-                    'envs', environment.id)
+                remoteWorkingRoot = executor.getRemoteExperimentRootDir(getExperimentId());
+                environment.runnerWorkingFolder = executor.joinPath(remoteWorkingRoot, 'envs', environment.id);
             }
-            environment.command = `cd ${environment.runnerWorkingFolder} && \
+            environment.command = `cd ${remoteWorkingRoot} && \
```

Comment:
why change to `remoteWorkingRoot`?

Reply:
because `${environment.runnerWorkingFolder}` is `./exp_id/env/runner_id` and we have ``${environment.command}=`mkdir -p envs/${envId} && cd envs/${envId} && ${environment.command}` `` in trialDispatcher.py, so change dir to `./exp_id`.

Reply:
command also contains `sh ../install_nni.sh`, if you only change root folder of the script, it can not find `install_nni.sh` file. I have another PR to fix this: https://github.com/microsoft/nni/pull/3472, perhaps we could merge the logic.